### ISSUE TITLE
Installed OK bot no Wifi Card Detected

### ISSUE THREAD

**Issue by matiasgatti** [tomaspinho/rtl8821ce#353]
Hi, I had a new Laptop Asus Vivobook 15. Installed Ubuntu 24.04 and wifi works bad: I got disconnections and errors of all kind. So, I decided to install RTL8821CE, installation went OK. But can get wifi card work with this driver: if I blacklist **rtw88_8821ce** in /etc/modprobe.d/blacklist.conf  wifi card is not detected.

Info:

uname -r

`6.8.0-31-generic
`

```
sudo dkms status
rtl8821ce/v5.5.2_34066.20200325, 6.8.0-31-generic, x86_64: installed
```

lsmod | grep 8821
```
8821ce               2514944  0
cfg80211             1339392  3 mac80211,rtl8xxxu,8821ce
```

lspci -k

```
0000:02:00.0 Network controller: Realtek Semiconductor Co., Ltd. RTL8821CE 802.11ac PCIe Wireless Network Adapter
	DeviceName: WLAN
	Subsystem: AzureWave RTL8821CE 802.11ac PCIe Wireless Network Adapter
	Kernel modules: rtw88_8821ce, 8821ce

```

sudo dmesg | grep 8821
```
[    2.153843] 8821ce: loading out-of-tree module taints kernel.
[    2.153880] 8821ce: module verification failed: signature and/or required key missing - tainting kernel
[    2.329255] rtl8821ce 0000:02:00.0: enabling device (0000 -> 0003)

```
(Secure Boot is Disabled)

If I remove "blacklist rtw88_8821ce" I have wifi with kernel driver. But got a lot of errors/disconnetions.

**Comment by dubhater** [comment]
> [    2.329255] rtl8821ce 0000:02:00.0: enabling device (0000 -> 0003)

I think that line indicates that the driver is detecting and handling your wifi card. If it's not working well, that's unfortunately because no one is going to fix it.

The problem you have with rtw88 may have been fixed already. Depends what the problem is. You can try a newer version of it: https://github.com/lwfinger/rtw88/

**Comment by mehdi-wsc** [comment]
@matiasgatti  I bought Asus yesterday and I'm facing exactly the same issue, the wifi card is not detected at all, I've installed the firmware but it's not helping
Can you tell me how did you solve it ? 
Thank you

**Comment by matiasgatti** [comment]
> @matiasgatti I bought Asus yesterday and I'm facing exactly the same issue, the wifi card is not detected at all, I've installed the firmware but it's not helping Can you tell me how did you solve it ? Thank you

Finally. I installed Ubuntu 22.04 lts. And i use the 'ubuntu driver' .sometimes i have some disconnects.. but works pretty good..

**Comment by dubhater** [comment]
@mehdi-wsc Are you sure your wifi card is RTL8821CE?  What does "lspci -nn" say?

**Comment by mehdi-wsc** [comment]
I was wrong, my network card is not a realtk but  a mediatek device 7902.
Any help with this card please 🙏?

**Comment by dubhater** [comment]
In that case, you are out of luck. Lots of people are asking about a Linux driver for MT7902 but no one found it.

If it's removable, you can try to replace it with another card. But your laptop may not accept it.

Here you can find a list of USB adapters that should work in Linux: https://github.com/morrownr/USB-WiFi

**Comment by boustanihani** [comment]
Having the same issue here, this is how I installed `rtl8821ce` this using NixOS 24.11:

```
  boot.blacklistedKernelModules = [ "rtw88_8821ce" ];

  boot.extraModulePackages = with config.boot.kernelPackages; [ rtl8821ce ];

  boot.kernelModules = [ "8821ce" ];

  hardware.enableRedistributableFirmware = true;
```

Am I missing something?

**Comment by NightSling** [comment]
Same with me!

**Comment by dubhater** [comment]
@boustanihani @NightSling What does "lspci -nn" say?

**Comment by NightSling** [comment]
> @boustanihani @NightSling What does "lspci -nn" say?

It was the same as the person above. 

The module loaded completely fine; It's just that the wifi card was not detected. 

Rfkill didn't show the wifi, only Bluetooth. (Unblocked)

Iwd and Nm also failed to fetch the adapter information. 

For now, I have moved to Fedora for a more stable experience.

**Comment by boustanihani** [comment]
In my case the module was also loaded fine; only the wifi card was not detected..

@NightSling Why Fedora? Does your card work on Fedora without any issues?

**Comment by boustanihani** [comment]
I also tried `Bluefin Linux` (based on Fedora 42) using latest kernel (`v6.14.5`) and I am still having the same issues. The network card suddenly disappears, and sometimes it also causes the system to freeze, Adding `pci=noaer` to the kernel boot params fixed the freezing issue, but didn't prevent the wifi card from suddenly disappearing..

**Comment by dubhater** [comment]
@boustanihani Try to update the computer's firmware.

**Comment by boustanihani** [comment]
I tried, no updates were detected..

**Comment by amlcx** [comment]
I'm in the same situation. Installation is fine, but even doing `sudo modprobe 8821ce` doesn't seem to have any effect, `lspci -k` shows that the driver is not in use.

**Comment by amlcx** [comment]
Coming back to this, what did the trick for me was rebuilding initramfs. The command varies for each distro.

**Comment by dubhater** [comment]
@amlcx Just curious, what driver is lspci -k showing now?

**Comment by amlcx** [comment]
@dubhater 

```
01:00.0 Network controller: Realtek Semiconductor Co., Ltd. RTL8821CE 802.11ac PCIe Wireless Network Adapter
        Subsystem: Hewlett-Packard Company Device 884d
        Kernel driver in use: rtl8821ce
        Kernel modules: rtw88_8821ce, 8821ce
```
